plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-request.php

```diff
@@ -88,6 +88,211 @@ public function get_request_url( $order, $sandbox = false ) {
 		return $this->endpoint . http_build_query( $paypal_args, '', '&' );
 	}
 
+	/**
+	 * Create a PayPal order using the Orders v2 API.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return array|null
+	 * @throws Exception If the PayPal order creation fails.
+	 */
+	public function create_paypal_order( $order ) {
+		try {
+			$request_body = $this->get_create_paypal_order_request_body( $order );
+
+			// phpcs:disable Generic.Commenting.Todo.TaskFound
+			// TODO: Replace with the wpcom endpoint when it's ready.
+			$response      = wp_remote_post(
+				$this->get_paypal_create_order_request_url(),
+				array(
+					'method'  => 'POST',
+					'headers' => array(
+						'Content-Type' => 'application/json',
+					),
+					'body'    => wp_json_encode( $request_body ),
+				)
+			);
+			$http_code     = wp_remote_retrieve_response_code( $response );
+			$body          = wp_remote_retrieve_body( $response );
+			$response_data = json_decode( $body, true );
+
+			if ( ! in_array( $http_code, array( 200, 201 ), true ) ) {
+				throw new Exception( 'PayPal order creation failed. Response status: ' . $http_code );
+			}
+
+			// Find the 'approve' link in the response -- this is where we will redirect the customer to.
+			$redirect_url = null;
+			foreach ( $response_data['links'] as $link ) {
+				if ( 'approve' === $link['rel'] && 'GET' === $link['method'] ) {
+					$redirect_url = $link['href'];
+					break;
+				}
+			}
+
+			return array(
+				'id'           => $response_data['id'],
+				'redirect_url' => $redirect_url,
+			);
+		} catch ( Exception $e ) {
+			WC_Gateway_Paypal::log( $e->getMessage() );
+			return null;
+		}
+	}
+
+	/**
+	 * Get the approve link from the response data.
+	 *
+	 * @param int   $http_code The HTTP code of the response.
+	 * @param array $response_data The response data.
+	 * @return string|null
+	 */
+	private function get_approve_link( $http_code, $response_data ) {
+		// See https://developer.paypal.com/docs/api/orders/v2/#orders_create.
+		$rel = ( 201 === $http_code ) ? 'approve' : 'payer-action';
+
+		foreach ( $response_data['links'] as $link ) {
+			if ( $rel === $link['rel'] && 'GET' === $link['method'] ) {
+				return $link['href'];
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Get the PayPal create-order request URL.
+	 *
+	 * @return string
+	 */
+	private function get_paypal_create_order_request_url() {
+		// phpcs:ignore Generic.Commenting.Todo.TaskFound
+		// TODO: This will be replaced with a constant pointing to the wpcom endpoint.
+		return get_site_url( null, 'wp-json/wc/v3/paypal-proxy/create-order' );
+	}
+
+	/**
+	 * Build the request body for the PayPal create-order request.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return array
+	 */
+	private function get_create_paypal_order_request_body( $order ) {
```

Comment:
```suggestion
	private function get_paypal_create_order_request_body( $order ) {
```

This sounds better meaningful, similar to how you named the `get_paypal_create_order_request_url` method.

Reply:
Updated, to be consistent with the other method!